src/services/transaction_watcher_signer_service.ts

```diff
@@ -192,14 +198,33 @@ export class TransactionWatcherSignerService {
         const latestBlockTimestamp = await this._web3Wrapper.getBlockTimestampAsync('latest');
         return new Date(latestBlockTimestamp * ONE_SECOND_MS);
     }
-    private async _signAndBroadcastTransactionsAsync(): Promise<void> {
+    private async _cancelOrSignAndBroadcastTransactionsAsync(): Promise<void> {
         const unsignedTransactions = await this._transactionRepository.find({
             where: [{ status: TransactionStates.Unsubmitted }],
         });
         logger.trace(`found ${unsignedTransactions.length} transactions to sign and broadcast`);
         for (const tx of unsignedTransactions) {
-            const signer = await this._getNextSignerAsync();
-            await this._signAndBroadcastMetaTxAsync(tx, signer);
+            if (TransactionWatcherSignerService._isUnsubmittedTxExpired(tx)) {
+                logger.error({
+                    message: `found a transaction in an unsubmitted state waiting longer that ${TX_HASH_RESPONSE_WAIT_TIME_MS}ms`,
+                    refHash: tx.refHash,
+                    from: tx.from,
+                });
+                tx.status = TransactionStates.Cancelled;
```

Comment:
non-blocking: do you think aborted makes sense here? Where aborted is we either kicked it out or it was way expired? Not sure if we need the granularity and they both feel `Aborted/Cancelled`.

Reply:
The distinction would help to quickly identify the transactions that were simply unsigned by the `TransactionWatcherSigner` because of error / low balance / transaction rejection etc.

`Aborted` transactions currently refer specifically to transactions which were superseded by a transaction with a higher gas price and the same nonce.